buildlib/az-helpers.sh

```diff
@@ -176,8 +176,8 @@ try_load_cuda_env() {
     if [ -f "/proc/driver/nvidia/version" ]; then
         have_cuda=yes
         have_gdrcopy=yes
-        az_module_load dev/cuda11.4 || have_cuda=no
-        az_module_load dev/gdrcopy2.3_cuda11.4 || have_gdrcopy=no
+        az_module_load dev/cuda12.2.2 || have_cuda=no
```

Comment:
Why don't we use latest and greatest CUDA 12.3.1?

Reply:
The version was specifically requested by the customer.